data-access-layer/eventmesh/ApiServerClient.js

```diff
@@ -669,6 +756,246 @@ class ApiServerClient {
       })
       .then(resource => _.get(resource, 'spec.options.context'));
   }
+
+  /**
+   * @description Create OSB Resource in Apiserver with the opts
+   * @param {string} opts.resourceGroup - Name of resource group 
+   * @param {string} opts.resourceType - Type of resource 
+   * @param {string} opts.resourceId - Unique id of resource 
+   * @param {string} opts.metadata - Optional; pass finalizers or some other field
+   * @param {string} opts.labels - to be put in label
+   * @param {Object} opts.spec - Value to set for spec field of resource
+   * @param {string} opts.status - status of the resource
+   */
+  createOSBResource(opts) {
+    logger.info(`Creating resource with opts: `, opts);
+    assert.ok(opts.resourceGroup, `Property 'resourceGroup' is required to create resource`);
+    assert.ok(opts.resourceType, `Property 'resourceType' is required to create resource`);
+    assert.ok(opts.resourceId, `Property 'resourceId' is required to create resource`);
+    assert.ok(opts.spec, `Property 'spec' is required to create resource`);
+    const metadata = _.merge(opts.metadata, {
+      name: opts.resourceId
+    });
+    if (opts.labels) {
+      metadata.labels = opts.labels;
+    }
+    const crdJson = this.getCrdJson(opts.resourceGroup, opts.resourceType);
+    const resourceBody = {
+      apiVersion: `${crdJson.spec.group}/${crdJson.spec.version}`,
+      kind: crdJson.spec.names.kind,
+      metadata: metadata,
+      spec: camelcaseKeys(opts.spec)
+    };
+
+    if (opts.status) {
+      const statusJson = {};
+      _.forEach(opts.status, (val, key) => {
+        if (key === 'state') {
+          resourceBody.metadata.labels = _.merge(resourceBody.metadata.labels, {
+            'state': val
+          });
+        }
+        statusJson[key] = _.isObject(val) ? JSON.stringify(val) : val;
+      });
+      resourceBody.status = statusJson;
+    }
+    // Create Namespace if not default
+    const namespaceId = this.getNamespaceId(opts.resourceType === CONST.APISERVER.RESOURCE_TYPES.INTEROPERATOR_SERVICEBINDINGS ?
+      _.get(opts, 'spec.instance_id') : opts.resourceId
+    );
```

Comment:
No this is different. In case of `SFServiceBinding` resource, we create it in the `SFServiceInstance` 's namespace. Hence the `namespaceId` is _.get(opts, 'spec.instance_id') for the same.

Reply:
Can we not set **instance_id** in params like in [putBinding](https://github.com/cloudfoundry-incubator/service-fabrik-broker/blob/4abf4e2aaa16feefd2e264abc5818cd8e33cc3f6/api-controllers/ServiceBrokerApiController.js#L331) and remove this condition. I feel it will be more readable.